src/xnatdcat/xnat_parser.py

```diff
@@ -158,3 +158,57 @@ def xnat_to_RDF(session: XNATSession, config: Dict) -> Graph:
         logger.warning("There were %d projects with invalid data for DCAT generation", failure_counter)
 
     return export_graph
+
+
+def split_keywords(xnat_keywords: Union[str, None]) -> List[str]:
+    """Takes an XNAT keyword list and splits it up into a list of keywords
+
+    Removes periods, colons and semicolons from the keywords
+
+    Parameters
+    ----------
+    xnat_keywords : str
+        String containing all keywords (space-separated)
+
+    Returns
+    -------
+    List[str]
+        List of keywords, empty list if there are no keywords
+    """
+    keyword_list = []
+    if xnat_keywords:
+        if len(xnat_keywords.strip()) > 0:
+            keyword_list = [
+                kw.strip()
+                for kw in xnat_keywords.strip().replace(".", " ").replace(",", " ").replace(";", " ").split(" ")
+            ]
+
+    # Filter out all empty strings, then return list
+    return list(filter(None, keyword_list))
+
+
+def _check_optin_optout(project, config: Dict) -> bool:
+    """This function checks if the project is elligible for indexing, given the opt-in/opt-out keywords
+
+    Parameters
+    ----------
+    project : XNAT
+        XNAT project of which the elligiblity needs to be determined
+    config : Dict
+        Configuration dictionary with the opt-in/opt-out keys
+
+    Returns
+    -------
+    bool
+        Returns True if a project is elligible for indexing, False if it is not.
+    """
+    if config.get('xnatdcat').get('optin'):
```

Comment:
is it possible 'xnatdcat' key is missing? then it will fail with Nonetype object does not have attribute 'get' on second get. If it is not possible - why do you need get here? Looks like parameters are set in cli_app.py so why not compare with None or just check if the parameter is in keywords directly?
General: is it possible that you would like to define optin/out as lists not a single keyword?

Reply:
It's possible, but means configuration file is badly formatted (that should be follow-up story, validator for config file).
I redid the code a bit to make behavior more clear and handle the edge case. Please take a look.

Multiple keywords can be considered, but for now I'd say it is out-of-scope.